src/Service/AbstractService.php

```diff
@@ -11,26 +11,82 @@
 use ShipEngine\ShipEngineClient;
 
 /**
- * Serialize and send HTTP requests.
+ * Serialize and send RPC requests over HTTP messages.
  */
 abstract class AbstractService
 {
+    /**
+     * @var ShipEngineClient
+     */
     protected ShipEngineClient $client;
+
+    /**
+     * @var MessageFactory
+     */
     protected MessageFactory $message_factory;
 
+    /**
+     * @var string
+     */
+    private const JSON_RPC_SPEC = '2.0';
```

Comment:
You can probably get rid of this and RPC_PATH unless they are going to be used in multiple places.